redaxo/src/addons/install/lib/api_package_upload.php

```diff
@@ -30,14 +30,18 @@ public function execute()
         try {
             if ($upload['upload_file']) {
                 $archive = rex_path::addonCache('install', md5($addonkey . time()) . '.zip');
-                $exclude = [];
+                $excludeDirs = [];
+                $excludeFiles = [];
                 if ($upload['replace_assets']) {
-                    $exclude[] = 'assets';
+                    $excludeDirs[] = 'assets';
                 }
                 if ($upload['ignore_tests']) {
-                    $exclude[] = 'tests';
+                    $excludeDirs[] = 'tests';
                 }
-                rex_install_archive::copyDirToArchive(rex_path::addon($addonkey), $archive, null, $exclude);
+                $excludeDirs[] = '.git';
+                $excludeDirs[] = 'node_modules';
```

Comment:
Hier bin ich unsicher ob wir das als default reinnehmen sollten... ich kann mir vorstellen, dass es addon gibt, die bewusst den ordner enthalten um js libraries zu bundeln

Reply:
Eigentlich ist es nicht praktikabel, in Production aus den node_modules zu konsumieren. Sie landen normalerweise auch gar nicht mit auf dem Server. Ich hoffe nicht, dass irgendein AddOn das macht — aktuell arbeiten kaum welche überhaupt mit Package Managern fürs Frontend —, und wenn doch, sollte es sich im eigenen Interesse umstellen.

Reply:
Gerade mal durch FOR gewühlt: Dort kommt es nicht vor. Die 3 AddOns von Thomas Goellner benutzen nur devDependencies.

Reply:
Bin da auch kein Experte, glaube aber dass man bei NodeJS serverseitig den ordner bräuchte.
Die Frage ist, ist das realistisch, dass ein REDAXO-Addon serverseitig mit NodeJS arbeitet? Keine Ahnung.

Wegen mir dürfte es so pauschal ausgeschlossen werden. Wäre aber genauso einverstanden, es konfigurierbar in der package.yml zu machen, welche Ordner ignoriert werden sollen.